Add EnelGrid integration documentation

sathia-musso opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).

  Before submitting your pull request, please verify that you have chosen the correct target branch,
  and that the PR preview looks fine and does not include unrelated changes.
-->
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [x] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to related core PR: [home-assistant/core#142742](https://github.com/home-assistant/core/pull/142742)  
- Link to parent pull request in the Brands repository: [home-assistant/brands#6873](https://github.com/home-assistant/brands/pull/6873)  
- This PR fixes or closes issue: N/A



## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **New Features**
  - Introduced the EnelGrid integration for Home Assistant, enabling monitoring of electricity consumption from Enel Italia.
  - Displays hourly energy data, daily usage, and cumulative monthly consumption on the Energy Dashboard.
  - Supports automatic daily updates and secure re-authentication.

- **Documentation**
  - Provides clear setup instructions, including required account details and pricing configuration.
  - Includes visual examples and troubleshooting tips for seamless integration setup.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

c0ffeeca7 commented:
Hi there, @sathia-musso :wave: 

Thank you for contributing.

1. As this PR seems to be related to a code PR, please rebase and point this PR against the `next` branch.
2. In the PR template, add the link to the related core PR.
3. Please find the documentation on contributing integration documentation in the developer docs:
    - [documentation standard](https://developers.home-assistant.io/docs/documenting/standards)
    - [How to create a new integration page, and what goes into the page header](https://developers.home-assistant.io/docs/documenting/create-page)
    - [documentation structure](https://developers.home-assistant.io/docs/documenting/integration-docs-examples)

c0ffeeca7 commented:
I'm [drafting this PR](https://developers.home-assistant.io/docs/review-process#prs-are-being-drafted-when-changes-are-needed), as changes are needed. Please undraft it, once the changes have been implemented and the PR is ready for review again.

Commit:
Add EnelGrid integration documentation

c0ffeeca7 commented on source/_integrations/enelgrid.markdown:
```diff
@@ -0,0 +1,79 @@
+| Automatic Daily Fetch              | ✅         |
+| Re-authentication Support          | ✅         |
+
+## Additional information
+
+- **Link to related core PR:** [home-assistant/core#142742](https://github.com/home-assistant/core/pull/142742)  
+- **Link to parent pull request in the Brands repository:** [home-assistant/brands#6873](https://github.com/home-assistant/brands/pull/6873)  
+- This PR fixes or closes issue: N/A
```
The information you listed under "Additional information" does not belong in the user documentation; it belongs in the PR template.
Please read the documentation on how to create integration documentation that was provided here:
https://github.com/home-assistant/home-assistant.io/pull/38457#issuecomment-2796714312

c0ffeeca7 commented on source/_integrations/enelgrid.markdown:
```diff
@@ -0,0 +1,79 @@
+## 🔗 External Links
+
+- [Enel Portal](https://www.enel.it/)
+- [Home Assistant Integrations](https://www.home-assistant.io/integrations/)
```
```suggestion
```
Links to related topics can be added to the header. Please follow the guideline on this.

c0ffeeca7 commented on source/_integrations/enelgrid.markdown:
```diff
@@ -0,0 +1,79 @@
+## 👤 Author
+
+Developed by [Sathia Francesco Musso](https://github.com/sathia-musso), Contributions welcome!
+
```
```suggestion
```
This information is part of the header, under codeowner.

c0ffeeca7 commented on source/_integrations/enelgrid.markdown:
```diff
@@ -0,0 +1,79 @@
+  - config_flow
+  - has_entity_name
+  - entity_unique_id
+  - appropriate_polling
```
```suggestion
```
where did you get the documentation category from? What does it do? I don't think this is a component of the header. Please refer to the documentation.
https://github.com/home-assistant/home-assistant.io/pull/38457#issuecomment-2796714312

sathia-musso committed:
Update source/_integrations/enelgrid.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

sathia-musso committed:
Update source/_integrations/enelgrid.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

sathia-musso committed:
Update source/_integrations/enelgrid.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

sathia-musso committed:
Merge branch 'next' into enelgrid-docs

sathia-musso committed:
Update enelgrid.markdown

sathia-musso committed:
Merge branch 'next' into enelgrid-docs

sathia-musso committed:
Update enelgrid.markdown

Add initial user documentation for EnelGrid integration, including setup instructions, Energy Dashboard guide, and supported features + nitpicking

sathia-musso committed:
Update enelgrid.markdown

nitpick

c0ffeeca7 commented on source/_integrations/enelgrid.markdown:
```diff
@@ -0,0 +1,63 @@
+---
+title: EnelGrid
+description: Integration for Enel Grid consumption data in Home Assistant
+stage: integration
```
```suggestion
```

c0ffeeca7 commented on source/_integrations/enelgrid.markdown:
```diff
@@ -0,0 +1,63 @@
+iot_class: cloud_polling
+integration_type: service
+dependencies:
+  - recorder
```
```suggestion
```

c0ffeeca7 commented on source/_integrations/enelgrid.markdown:
```diff
@@ -0,0 +1,63 @@
+ha_integration_type: integration
+---
+
+The **EnelGrid** integration connects Home Assistant with **Enel Italia**, allowing users to import and monitor their electricity consumption data.
```
```suggestion
The **EnelGrid** {% term integrations %} connects Home Assistant with **Enel Italia**, allowing users to import and monitor their electricity consumption data.
```
we have a [glossary term reference](https://developers.home-assistant.io/docs/documenting/integration-docs-examples#glossary-term-reference) for this

c0ffeeca7 commented on source/_integrations/enelgrid.markdown:
```diff
@@ -0,0 +1,63 @@
+- Automatically updates daily
+- Supports secure re-authentication if needed
+
+## Configuration
```
```suggestion
## Prerequisites
```

c0ffeeca7 commented on source/_integrations/enelgrid.markdown:
```diff
@@ -0,0 +1,63 @@
+- **Price per kWh**: Estimated based on your billing rate (total cost ÷ total kWh consumed)
+
+All credentials are stored securely by Home Assistant’s configuration system.
+
```
```suggestion
{% include integrations/config_flow.md %}

```
If the integration supports config flow, there a [reusable text block]({% include integrations/config_flow.md %}) you can use for the Configuration section.

sathia-musso committed:
Update enelgrid.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

sathia-musso committed:
Update enelgrid.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

sathia-musso committed:
Update enelgrid.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

sathia-musso committed:
Update enelgrid.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

sathia-musso committed:
Update source/_integrations/enelgrid.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>
